tutorials/tree/dictionary/readTree.cpp

```diff
@@ -0,0 +1,48 @@
+// Author: Alvaro Tolosa-Delgado CERN 07/2023
+// Author: Jorge Agramunt Ros IFIC(Valencia,Spain) 07/2023
+
+/*************************************************************************
+ * Copyright (C) 1995-2023, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <iostream>
+
+#include <TFile.h>
+#include <TTree.h>
+#include <TTreeReader.h>
+#include <TTreeReaderValue.h>
+
+#include "data2Tree.hpp"
+
+void readTree()
+{
+
+   std::unique_ptr<TFile> ifile { TFile::Open("testFile.root", "read") };
+   if ( !ifile || ifile->IsZombie() )  {
+      std::cerr << "Problem opening the file testFile.root" << std::endl;
+      return;
+   }
+
+   // Create a TTreeReader to read the tree named "myTree"
+   TTreeReader aReader("myTree", ifile.get() );
+
+   // Create TTreeReaderValues for the branches "branch1" and "branch2"
+   TTreeReaderValue<myDetectorData> branch1(aReader, "branch1.");
+   TTreeReaderValue<myDetectorData> branch2(aReader, "branch2.");
+
+   // Loop over the entries of the tree
+   while (aReader.Next()) {
+      if (branch1->time != 0)
+         std::cout << " -Branch1 : time: " << branch1->time << "\t energy: " << branch1->energy << std::endl;
+      else if (branch2->time != 0)
+         std::cout << " +Branch2 : time: " << branch2->time << "\t energy: " << branch2->energy << std::endl;
+      else
+         std::cerr << "WARNING: entry " << aReader.GetCurrentEntry() << " is empty! " << std::endl;
+   }
+
+}
+
```

Comment:
I understand that for the writing part of the tutorial we want to show usage of bare TTree, but for the processing I believe we should really only show RDataFrame at this point. I don't see any reason to use TTreeReader as an analysis interface here.

I would suggest reframing this part of the tutorial with RDataFrame. Something akin to this should be a good starting point
```
ROOT::RDataFrame df{"myTree", "testFile.root"};
df.Display({"branch1.time","branch1.energy","branch2.time","branch2.energy"}, /*nRows*/ 10)->Print();
```